frontends/main/src/app-pages/HomePage/VideoShortsModal.tsx

```diff
@@ -0,0 +1,246 @@
+        if (isPlaying(videosRef.current[selectedIndex])) {
+          videosRef.current[selectedIndex]?.pause()
+        } else {
+          videosRef.current[selectedIndex]?.play().catch(() => {})
+        }
+      }
+
+      event.preventDefault()
+    }
+
+    document.addEventListener("keydown", handleKeyDown)
+    return () => {
+      document.removeEventListener("keydown", handleKeyDown)
+    }
+  }, [onClose, selectedIndex])
+
+  const onSlidesInView = (inView: number[]) => {
+    if (inView.length === 1) {
+      videosRef.current
+        .filter((video, index) => video && index !== inView[0])
+        .forEach((video) => {
+          video!.pause()
+        })
+      setSelectedIndex(inView[0])
+      if (videosRef.current[inView[0]]) {
+        const video = videosRef.current[inView[0]]!
+        video.muted = muted
+
+        // On iOS, only autoplay if muted or if user has interacted
+        if (!isIOS() || muted || hasUserInteracted) {
+          video.play().catch(() => {})
+        }
+      }
+    }
+  }
+
+  const onClickMute = () => {
+    if (selectedIndex !== null && videosRef.current[selectedIndex]) {
+      const video = videosRef.current[selectedIndex]!
+      const wasMuted = video.muted
+      video.muted = !wasMuted
+
+      setHasUserInteracted(true)
+
+      if (wasMuted && !video.paused) {
+        video.play().catch(() => {})
+      }
+    }
+    setMuted(!muted)
+  }
+
+  const handleVideoClick = () => {
+    if (selectedIndex !== null && videosRef.current[selectedIndex]) {
+      const video = videosRef.current[selectedIndex]!
+      setHasUserInteracted(true)
+
+      if (video.paused) {
+        video.play().catch(() => {})
+      } else {
+        video.pause()
+      }
+    }
+  }
+
+  return (
+    <Overlay>
+      <CloseButton size="large" edge="rounded" variant="text" onClick={onClose}>
+        <RiCloseLine />
+      </CloseButton>
+      <MuteButton
+        size="large"
+        edge="rounded"
+        variant="text"
+        onClick={onClickMute}
+      >
+        {muted ? <RiVolumeMuteLine /> : <RiVolumeUpLine />}
+      </MuteButton>
+      <CarouselV2Vertical
+        initialSlide={startIndex}
+        onSlidesInView={onSlidesInView}
+      >
+        {videoData?.map((item: VideoShort, index: number) => (
+          <CarouselSlide
+            key={index}
+            width={(height - 60) * (9 / 16)}
+            data-index={index}
+          >
+            {selectedIndex !== null && Math.abs(selectedIndex - index) < 2 ? (
+              <Video
+                ref={(el) => {
+                  if (videosRef.current && el) {
+                    videosRef.current[index] = el
+                    el.addEventListener("error", (e: Event) => {
+                      console.error("Video error:", e)
+                    })
+                  }
+                }}
+                onClick={handleVideoClick}
+                // TODO: Using a temporary bucket on GCP owned by jk
```

Comment:
Is this something that will be resolved in a follow up or can we move this to s3?